### PR TITLE
chore: Bump DOCA Driver to `24.04-0.4.0.0-0`

### PR DESCRIPTION
**Pull request by killianmuldoon**
Bump DOCA Driver to 24.04-0.4.0.0-0

Signed-off-by: killianmuldoon <kmuldoon@nvidia.com>

### CODE REVIEW

**Comment by rollandf** [comment]
@killianmuldoon please rebase

**Review by rollandf — DISMISSED** [review]

File: hack/release.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ SriovIbCni:
 Mofed:
   image: doca-driver
   repository: nvcr.io/nvstaging/mellanox
-  version: 24.04-0.3.0.0-0
+  version: 24.01-0.3.3.1-8
```
Comment (e0ne) [inline hack/release.yaml:31]: we need to use 24.04-x.x.x.x-x build

**Review by rollandf — APPROVED** [review]

**Review by e0ne — APPROVED** [review]

**Pull request merged**